analyses/chromosomal-instability/00-setup-breakpoint-data.R

```diff
@@ -355,10 +356,18 @@ breaks_density_list <- lapply(breaks_list, function(breaks_df) {
     dplyr::group_by(
       samples, experimental_strategy, genome_size
     ) %>%
-    # Count number of mutations for that sample
-    dplyr::summarize(breaks_count = dplyr::n()) %>%
-    # Calculate breaks density
-    dplyr::mutate(breaks_density = breaks_count / (genome_size / 1000000))
+    # Count number of mutations for that sample but find out if it is NA
+    dplyr::summarize(is_na = any(is.na(chrom)), 
+                     breaks_count = dplyr::n()) %>%
+    # Calculate breaks density, but put NA for breaks_count if the sample was not 
+    # in the SV or CNV data originally
+    dplyr::mutate(breaks_count = dplyr::case_when(
+      !is_na ~ as.numeric(breaks_count), 
+      is_na ~ as.numeric(NA)
+      ), 
```

Comment:
Unfortunately, this isn't actually doing what we really want it to do. 
The trouble is that anything with a true zero count will get an NA in chrom column, just like something that is actually missing. 

I think the thing to do is to add a column to `metadata` called `surveyed` which would indicate if the sample is in  `unique(c(cnv_samples, sv_samples))`. Then if `surveyed` is TRUE, set the `is_na` `break_counts` to 0, and to NA otherwise.

something like:
```r
dplyr::mutate(
    breaks_count = dplyr::case_when(
      !is_na ~ as.numeric(breaks_count), 
      is_na & surveyed ~ as.numeric(0),
      TRUE ~ as.numeric(NA)
      ), 
```